### PR TITLE
add init system to default grains

### PR DESCRIPTION
**Pull request by cgtx**
I would find it highly useful to have a default grain describing a minion's init system.

``` jinja
{% if grains['init'] == 'systemd' %}
```

### CODE REVIEW

**Comment by steverweber** [comment]
i like this idea but please make the 'init' grain an array.

**Comment by steverweber** [comment]
well the code looks good... so never mind me :)

**Comment by terminalmage** [comment]
I like this. I made some optimizations in https://github.com/saltstack/salt/commit/2397e6c that will speed this up considerably.

File: salt/grains/core.py [hunk 1]
```diff
@@ -975,6 +975,23 @@ def os_data():
             grains['systemd']['version'] = systemd_info[0].split()[1]
             grains['systemd']['features'] = systemd_info[1]
 
+        # Add init grain
+        if os.path.isdir('/run/systemd/system'):
+            grains['init'] = 'systemd'
+        else:
```
Comment (kiorky) [inline salt/grains/core.py]: this lack docker support, with their .dockerinit thing

Comment (kiorky) [inline salt/grains/core.py]: cc @terminalmage

Comment (kiorky) [inline salt/grains/core.py]: there is also openrc, to check out

**Comment by morsik** [comment]
CentOS 6 reports:

<pre># strings /proc/1/cmdline 
/sbin/init</pre>

Code above doesn't looks like it should works for SystemV, cause it checks for `sysvinit`.

**Comment by cgtx** [comment]
@morsik That's not how it works.  It isn't running strings on `/proc/1/cmdline`, it is running strings on the file referenced by `/proc/1/cmdline`, which in this case is `/sbin/init`.

```
# cat /etc/redhat-release 
CentOS release 6.6 (Final)
# strings /sbin/init | grep -m1 upstart
upstart
```

Also note that @terminalmage [refactored that code](https://github.com/saltstack/salt/commit/2397e6c) to not need the actual strings command.

**Comment by gtmanfred** [comment]
@rallytime any chance this can get backported to 2015.5?

Thanks!

**Pull request merged**